x-pack/plugins/security_solution/public/resolver/store/reducer.ts

```diff
@@ -4,81 +4,97 @@
 import { cameraReducer } from './camera/reducer';
 import { dataReducer } from './data/reducer';
-import type { ResolverAction } from './actions';
-import type { ResolverState, ResolverUIState } from '../types';
+import type { AnalyzerState } from '../types';
 import { panAnimationDuration } from './camera/scaling_constants';
 import { nodePosition } from '../models/indexed_process_tree/isometric_taxi_layout';
+import {
+  appReceivedNewExternalProperties,
+  userFocusedOnResolverNode,
+  userSelectedResolverNode,
+  createResolver,
+  clearResolver,
+} from './actions';
+import { serverReturnedResolverData } from './data/action';
 
-const uiReducer: Reducer<ResolverUIState, ResolverAction> = (
-  state = {
-    ariaActiveDescendant: null,
-    selectedNode: null,
-  },
-  action
-) => {
-  if (action.type === 'serverReturnedResolverData') {
-    const next: ResolverUIState = {
-      ...state,
-      ariaActiveDescendant: action.payload.result.originID,
-      selectedNode: action.payload.result.originID,
-    };
-    return next;
-  } else if (action.type === 'userFocusedOnResolverNode') {
-    const next: ResolverUIState = {
-      ...state,
-      ariaActiveDescendant: action.payload.nodeID,
-    };
-    return next;
-  } else if (action.type === 'userSelectedResolverNode') {
-    const next: ResolverUIState = {
-      ...state,
-      selectedNode: action.payload.nodeID,
-      ariaActiveDescendant: action.payload.nodeID,
-    };
-    return next;
-  } else if (action.type === 'appReceivedNewExternalProperties') {
-    const next: ResolverUIState = {
-      ...state,
-      locationSearch: action.payload.locationSearch,
-      resolverComponentInstanceID: action.payload.resolverComponentInstanceID,
-    };
-    return next;
-  } else {
-    return state;
-  }
+export const initialAnalyzerState: AnalyzerState = {
+  analyzerById: {},
 };
 
-const concernReducers = combineReducers({
-  camera: cameraReducer,
-  data: dataReducer,
-  ui: uiReducer,
-});
-export const resolverReducer: Reducer<ResolverState, ResolverAction> = (state, action) => {
-  const nextState = concernReducers(state, action);
-  if (action.type === 'userSelectedResolverNode' || action.type === 'userFocusedOnResolverNode') {
-    const position = nodePosition(layout(nextState), action.payload.nodeID);
-    if (position) {
-      const withAnimation: ResolverState = {
-        ...nextState,
-        camera: animatePanning(
-          nextState.camera,
-          action.payload.time,
+const uiReducer = reducerWithInitialState(initialAnalyzerState)
+  .withHandling(
+    immerCase(createResolver, (draft, { id }) => {
+      if (!draft.analyzerById[id]) {
+        draft.analyzerById[id] = EMPTY_RESOLVER;
+      }
+      return draft;
+    })
+  )
+  .withHandling(
+    immerCase(clearResolver, (draft, { id }) => {
+      delete draft.analyzerById[id];
+      return draft;
+    })
+  )
+  .withHandling(
+    immerCase(
+      appReceivedNewExternalProperties,
```

Comment:
I think you should be able to do away with this. This was a way to get data from the main store into resolver's store/middleware.